components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -70,7 +72,11 @@ open class GleanInternalAPI {
         storageEngineManager = StorageEngineManager(applicationContext = applicationContext)
         pingMaker = PingMaker(storageEngineManager)
         this.configuration = configuration
-        httpPingUploader = HttpPingUploader(configuration)
+        if (BuildConfig.DEBUG) {
+            pingUploader = LogPingUploader(configuration)
+        } else {
+            pingUploader = HttpPingUploader(configuration)
+        }
```

Comment:
Note that `BuildConfig.DEBUG` means it is the `debug` variant of the library. Apps consuming the library from maven.mozilla.org will always get a release artifact and this path will never get triggered.
That's different for the sample app in the repository: It knows of the variants of the library and If you build the "debug" variant  of the app it will use a matching "debug" variant of the library.

Reply:
Got it.  So this is definitely not the way we want to do it.

Do you have any suggestions for creating a flag that developers could easily flip to turn this on/off?  It's not just as simple as just changing logging settings -- we also want to *not* send pings over Http when debugging ping content...

Reply:
That's a good question.

If you really want to attach this to the debug flag: You can read that at runtime. Here is an example: https://medium.com/@elye.project/checking-debug-build-the-right-way-d12da1098120

In the old telemetry library this was mostly left to the app and via the configuration the app could either:
* Enable/disable telemetry collection/uploading
* Or: Set a different uploader (that only logs instead of actually uploads)

I like your approach of hiding this behind a flag. I wonder if this is something you could just expose and leave it to the app to decide what builds need this setup?

Reply:
I would also slightly change the behaviour compared to the old telemetry a little bit: I would make it possible to dump pings to console in any case, without necessarily dropping the http uploader. This would allow easier debugging of weirder things on release. What do you think, @pocmo ?

Reply:
Yes, absolutely! And in addition to that: I would like to see events at the time they happen :)

Reply:
> 
> 
> Yes, absolutely! And in addition to that: I would like to see events at the time they happen :)

This could get a bit verbose, but we can consider doing it! It only takes a single log line :)

@mdboom , given the previous discussion, I think this PR can really boil down to just dump to logcat in the http uploader, in addition to uploading, if the configuration flag is true.